Assets/MixedRealityToolkit.SDK/Features/UX/Scripts/Sliders/PinchSlider.cs

```diff
@@ -33,20 +35,66 @@ public GameObject ThumbRoot
             }
         }
 
-        [Range(0, 1)]
+        #region MyValue 
+        [SerializeField]
+        private int minValue = -4;
+        [SerializeField]
+        private int maxValue = 4;
+
+        [SerializeField]
+        private bool wholeNumbers = true;
+
         [SerializeField]
+        private float startValue = 0f;
+
+        [Tooltip("Automatic placement of slider's steps.")]
+        [SerializeField]
+        private bool tickMarksAutoFill = true;
+
+        private float myValue = 0f;
+        public float MyValue
+        {
+            get { return myValue; }
+            set
+            {
+                var oldSliderValue = myValue;
+                myValue = value;
+                OnValueUpdated.Invoke(new SliderEventData(oldSliderValue, value, activePointer, this));
+            }
+        }
+
+        [Range(0, 1)]
         private float sliderValue = 0.5f;
         public float SliderValue
         {
             get { return sliderValue; }
             set
             {
                 var oldSliderValue = sliderValue;
-                sliderValue = value;
+                //Round value 
+                if (oldSliderValue != value)
+                {
+                    float temp = value * (maxValue - minValue);
+                    if (wholeNumbers == true)
+                    {
+                        int v = Mathf.RoundToInt(temp);
+                        if (MyValue != v + minValue)
+                        {
+                            MyValue = v + minValue;
+                            sliderValue = (float)v / ((float)maxValue - (float)minValue);
+                        }
+                    }
+                    else
+                    {
+                        MyValue = temp + minValue;
+                        sliderValue = temp / ((float)maxValue - (float)minValue);
```

Comment:
It is a bit interesting to me that a person could try to set sliderValue, but after doing this conversion to myValue and then back, it could actually end up being a different value (i.e. due to rounding errors when calculating temp and then going back)